### ISSUE TITLE
请教一下具体的Customize Models构造问题

### ISSUE THREAD

**Issue by ithok** [RUCAIBox/RecBole#1267]
首先Recbole很好用，感谢您（您和您团队的)付出！
因为我自己刚开始用这个框架，所以有一些具体的使用上的疑问想请教一下。
recbole确实在框架体系内复现了很多经典（前沿）算法，具体到我自身的疑问是我如何去使用这些模型，当然我了解就是如何通过recbole调用这些模型在特定数据集上运行。
但是涉及到Customize Models的时候，比如我现在是要自己设计一个model，recbole是否有提供一些可供使用的比较基础的构件呢？比如说GCN&GAT layer这种，使用dgl的时候可以自由组装来组网，但我不清楚recbole有没有提供清晰的类似api（因为这些构件在recbole里模型的复现代码里是可以看到的，但我也不清楚这些实现是不是都是统一的，比如GCMC就提供了GC encoder），还是原则上由使用者自己来实现这些组件呢？
已看过官方文档的Customize Models部分，以上就是全部问题。
非常感谢！

**Comment by hyp1231** [comment]
您好，感谢您的关注。

RecBole 是提供一些可复用的组建的，主要体现在两方面：
1. Customize Models 的开发的第一步是继承一类模型的基类，而这些基类一方面帮助协调这类模型的输入输出，另一方面本身就提供了一些可复用的模块和函数。如 `SequentialRecommender` 中提供了便于获取序列最后一个商品表示的 `gather_indexes`，以及获取 Transformer 需要的 attention mask 的 `get_attention_mask`。又如 `ContextRecommender` 提供了若干批量组织 feature 的函数。这部分可以关注 [abstract_recommender](https://github.com/RUCAIBox/RecBole/blob/master/recbole/model/abstract_recommender.py) 的代码。
2. 另外还有一些相对通用的可复用函数/类，如 `MLPLayers`, `TransformerEncoder` 等，放在 [layers](https://github.com/RUCAIBox/RecBole/blob/master/recbole/model/layers.py) 中。当然这里还是建议您在调用前阅读源码，因为毕竟推荐系统偏向应用，其自身比较有特色的技术相对较少，建议您了解实现模块内部的细节再考虑是否复用。

此外您也可以关注一些与您想实现的模型相关的实现，看是否可以从这些实现中进行复用。

这里也提供几个我们团队使用 RecBole 开发自定义模型的例子，仅供参考，[NCL](https://github.com/RUCAIBox/NCL) (general model) 和 [CORE](https://github.com/RUCAIBox/CORE) (sequential model)。

**Comment by ithok** [comment]
好的，感谢您的回复！
